game/engine/main.py

```
@@ -826,6 +930,27 @@ def run():
+            ep_row = int(next(tokens))
+            ep_col = int(next(tokens))
+            fr = int(next(tokens))
+            fc = int(next(tokens))
+            tr = int(next(tokens))
+            tc = int(next(tokens))
+            
+            promo = '\0'
+            try:
+                promo = next(tokens)
+            except StopIteration:
+                pass
```

Ruff (0.15.15)

[warning] Use `contextlib.suppress(StopIteration)` instead of `try`-`except`-`pass`

Replace `try`-`except`-`pass` with `with contextlib.suppress(StopIteration): ...`

(SIM105)